src/analysis/stack-lattice.h

```diff
@@ -0,0 +1,189 @@
+        modified = true;
+      }
+
+      return modified;
+    }
+
+    void print(std::ostream& os) {
+      if (!stackValue.has_value()) {
+        os << "TOP STACK" << std::endl;
+        return;
+      }
+
+      for (auto iter = stackValue->rbegin(); iter != stackValue->rend();
+           ++iter) {
+        iter->print(os);
+        os << std::endl;
+      }
+    }
+
+    friend StackLattice;
+  };
+
+  // Like in computing the LUB, we compare the suffixes of the two stacks.
+  // If the left stack is higher, and left suffix >= right suffix, then we say
+  // that left stack > right stack. If the left stack is lower an left suffix >=
+  // right suffix or if the left stack is higher and the right suffix > left
+  // suffix or they are unrelated, then there is no relation. Same applies for
+  // the reverse relationship.
+  LatticeComparison compare(const Element& left, const Element& right) {
+    if (left.isTop()) {
+      if (right.isTop()) {
+        return LatticeComparison::EQUAL;
+      } else {
+        return LatticeComparison::GREATER;
+      }
+    } else if (right.isTop()) {
+      return LatticeComparison::LESS;
+    }
+
+    bool hasLess = false;
+    bool hasGreater = false;
+
+    // Check the suffixes of both stacks. If there is an unrelated pair of
+    // elements, or if there is both a > and a <, then the stacks must be
+    // unrelated.
+    for (auto leftIterator = left.stackValue->crbegin(),
+              rightIterator = right.stackValue->crbegin();
+         leftIterator != left.stackValue->crend() &&
+         rightIterator != right.stackValue->crend();
+         ++leftIterator, ++rightIterator) {
+      LatticeComparison currComparison =
+        stackElementLattice.compare(*leftIterator, *rightIterator);
+      switch (currComparison) {
+        case LatticeComparison::NO_RELATION:
+          return LatticeComparison::NO_RELATION;
+          break;
+        case LatticeComparison::LESS:
+          hasLess = true;
+          break;
+        case LatticeComparison::GREATER:
+          hasGreater = true;
+          break;
+        default:
+          break;
+      }
+    }
+
+    // Check cases for when the stacks have unequal or equal heights.
+    // As a rule, if a stack is higher, it is greater than the other stack, but
+    // if and only if their suffixes have the same greater than or equal
+    // relationship.
+    if (left.stackValue->size() > right.stackValue->size()) {
+      hasGreater = true;
+    } else if (right.stackValue->size() > left.stackValue->size()) {
+      hasLess = true;
+    }
+
+    if (hasLess && !hasGreater) {
+      return LatticeComparison::LESS;
+    } else if (hasGreater && !hasLess) {
+      return LatticeComparison::GREATER;
+    } else if (hasLess && hasGreater) {
+      return LatticeComparison::NO_RELATION;
+    } else {
+      return LatticeComparison::EQUAL;
+    }
+  }
+
+  Element getBottom() {
+    Element result;
+    return result;
```

Comment:
```suggestion
    return {};
```

or

```suggestion
    return Element{};
```

would also work.